lib/index.js

```diff
@@ -1,16 +1,98 @@
 'use strict'
 
-var integration = require('@segment/analytics.js-integration')
-
+var integration = require('@segment/analytics.js-integration');
+var useHttps = require('use-https');
 /**
  * Expose `Wigzo` integration
  */
 
 var Wigzo = module.exports = integration('Wigzo')
+  .readyOnInitialize()
+  .global('wigzo')
+  .option('orgToken', '')
+  .option('features', [])
+  .tag('<script src="https://tracker.wigzopush.com/wigzo.js">')
+  .tag('https', '<link rel="manifest" href="/gcm_manifest.json">') /* This needs to be put inside <head>.Is this correct syntax ? */
+  .tag('https', '<script src="https://tracker.wigzopush.com/wigzopush_manager.js?orgtoken={{ orgToken }}">')
+  .tag('http', '<script src="https://tracker.wigzopush.com/gcm_http_subscribe.js?orgtoken={{ orgToken }}">');
 
 /**
  * Initialize Wigzo
  */
 
-Wigzo.prototype.initialize = function () {
-}
+Wigzo.prototype.initialize = function (){
+
+  window.wigzo.setIntegrationSource('Segment');
+
+  window.wigzo.init(settings.orgToken, settings.features);
+  var name = 'https';
+  if(!useHttps()){
+    name = 'http';
+    window.wigzo.httpGcmShowDialog = true;
+  } else {
+    window.wigzo.wigzoGcmAutoSubscribe = true;
+  }
+  
+  this.load(name, this.ready);
+};
+
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+
+Wigzo.prototype.loaded = function() {
+  return !!window.wigzo;
+};
+
+
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Identify} identify
+ */
+
+Wigzo.prototype.identify = function(identify) {
+  var traits = identify.traits();
+  var id = identify.userId();
+  if (id) window.wigzo.USERID = id;
+
+  window.wigzo.COOKIEID = identify.anonymousId;
+  if (traits) {
+    window.wigzo.identify({
+      email : traits.email ? traits.email : '',
+      phone : traits.phone ? traits.phone : '',
+      fullName : traits.name ? traits.name : ''
+    })
+  }
+};
+
+/**
+ * Track.
+ *
+ * @api public
+ * @param {Track} track
+ */
+
+Wigzo.prototype.track = function(track) {
+  var event = track.event();
+  var properties = track.properties();
+  window.wigzo.track(event, properties);
```

Comment:
i'd just make this all one line:

```
window.wigzo.track(track.event(), track.properties());
```

just want to confirm that your library can handle nested objects/arrays? since `properties` can be an arbitrary free dictionary of key/value pairs.